MainController.js

```diff
@@ -0,0 +1,33 @@
+// Code goes here
+
+(function() {
+
+  var app = angular.module("restaurantViewer");
+
+  var MainController = function($scope, $location) {
+
+    $scope.search = function(distance, latitude, longitude) {
+      url = "/search"
+
+      // Always put a slash for distance, so that if users type latitude and longitude but no distance, it looks
+      // like /search//X/Y instead of /search/X/Y, which would cause the search page to think distance is X,
+      // latitude is Y, and there's no longitude
+      if (distance != null) {
+        url += "/" + distance
+      } else {
+        url += "/"
+      }
```

Comment:
This is a little confusing, and it's because of the shitty routing for the search page that I talked about in my comment on `app.js`. I'd like to improve it.